disco/extensions/upgrade_simulation/upgrades/voltage_upgrade_functions.py

```diff
@@ -1259,76 +1301,175 @@ def plot_heatmap_distmatrix(square_array=None, fig_folder=None):
     plt.savefig(os.path.join(fig_folder, "Nodal_violations_heatmap.pdf"))
 
 
-def plot_feeder(G=None, show_fig=False, fig_folder=None):
+def plot_feeder(fig_folder, title, show_fig=False, circuit_source=None, enable_detailed=False):
+    """Function to plot feeder network.
+    """
+    G = generate_networkx_representation()
     position_dict = nx.get_node_attributes(G, 'pos')
     nodes_list = G.nodes()
+    Un_G = G.to_undirected()
 
     plt.figure(figsize=(7, 7))
     plt.figure(figsize=(40, 40), dpi=10)
     plt.clf()
-    ec = nx.draw_networkx_edges(G, pos=position_dict, alpha=1.0, width=0.3)
-    ldn = nx.draw_networkx_nodes(G, pos=position_dict, nodelist=nodes_list, node_size=2,
-                                 node_color='k')
-
-    # ld = nx.draw_networkx_nodes(G, pos=position_dict, nodelist=nodes_list, node_size=6,
-    #                             node_color='yellow', alpha=0.7)
-
-    # nx.draw_networkx_labels(G, pos=position_dict, node_size=1, font_size=15)
-    plt.title("Feeder with all customers having DPV systems")
-    plt.axis("off")
-    plt.savefig(os.path.join(fig_folder, "Feeder.pdf"))
+    nx.draw_networkx_edges(Un_G, pos=position_dict, alpha=1.0, width=0.3)
+    default_node_size = 2
+    default_node_color = 'black'
+    
+    NodeLegend = {
+        "Load": get_load_buses(dss), 
+        "PV": get_pv_buses(dss), 
+        "Transformer": list(get_all_transformer_info_instance(compute_loading=False)['bus_names_only'].str[0].values),
+    }
+    if circuit_source is not None:
+        NodeLegend["Circuit Source"] = [circuit_source]
+    if enable_detailed:
+        cap_df = get_capacitor_info()
+        if not cap_df.empty:
+            NodeLegend["Capacitor"] = list(cap_df['bus1'].str.split(".").str[0].unique())
+        reg_df =  get_regcontrol_info()
+        # if not reg_df.empty:
+    colored_nodelist = []
+    for key in NodeLegend.keys():
+        temp_list = NodeLegend[key]
+        colored_nodelist  = colored_nodelist + temp_list
+        if  len(temp_list) != 0:
+            nx.draw_networkx_nodes(Un_G, pos=filter_dictionary(dict_data=position_dict, wanted_keys=temp_list),
+                                nodelist=temp_list, node_size=NODE_COLORLEGEND[key]["node_size"], node_color=NODE_COLORLEGEND[key]["node_color"],
+                                alpha=NODE_COLORLEGEND[key]["alpha"], label=NODE_COLORLEGEND[key]["label"])
+    
+    remaining_nodes = list(set(nodes_list) - set(colored_nodelist)) 
+    nx.draw_networkx_nodes(Un_G, pos=filter_dictionary(dict_data=position_dict, wanted_keys=remaining_nodes),
+                                nodelist=remaining_nodes, node_size=default_node_size, node_color=default_node_color)
+    plt.title(title, fontsize=50)
+    # plt.axis("off")
+    plt.legend(fontsize=50)
+    plt.savefig(os.path.join(fig_folder, title+".pdf"))
     if show_fig:
         plt.show()
     return

```

Comment:
Did not capture before, may can be removed. This option is risky, thinking about run 1000 jobs with `show_fig=True`, 1000 figure windows would open.

Reply:
Oh, that makes sense.